snippets/card-product.liquid

```diff
@@ -336,21 +336,35 @@
     </div>
   </div>
 {%- else -%}
+  {%- liquid
+    assign ratio = 1
+    if media_aspect_ratio == 'portrait'
+      assign ratio = 0.8
+    elsif media_aspect_ratio == 'adapt'
+      assign ratio = card_product.featured_media.aspect_ratio
+    endif
+    if ratio == 0 or ratio == null
+      assign ratio = 1
+    endif
+  -%}
   <div class="card-wrapper product-card-wrapper underline-links-hover">
     <div
       class="
         card card--{{ settings.card_style }}
         {% if extend_height %} card--extend-height{% endif %}
+        {% if image_shape and image_shape != 'default' %} card--shape{% endif %}
         {% if settings.card_style == 'card' %} color-{{ settings.card_color_scheme }} gradient{% endif %}
       "
-      style="--ratio-percent: 100%;"
+      style="--ratio-percent: {{ 1 | divided_by: ratio | times: 100 }}%;"
     >
       <div
         class="card__inner{% if settings.card_style == 'standard' %} color-{{ settings.card_color_scheme }} gradient{% endif %} ratio"
-        style="--ratio-percent: 100%;"
+        style="--ratio-percent: {{ 1 | divided_by: ratio | times: 100 }}%;"
       >
-        <div class="card__media">
-          <div class="media media--transparent">
+        <div class="card__media {% if image_shape and image_shape != 'default' %} shape--{{ image_shape }} color-{{ settings.card_color_scheme }} gradient{% endif %}"
+        style="{% if  image_shape == 'default' and media_aspect_ratio == 'portrait' %} top: 3rem;{% endif %}">
+          <div class="media media--transparent"
+            style="{% if  image_shape != 'default' and media_aspect_ratio == 'portrait' %} top: 3rem;{% endif %}">
```

Comment:
Doesn't it work as expected when those `top: 3rem;` are removed ? 
Cause I see an issue with the product cards are set to a different color scheme than the section and with the card layout instead of standard 🤔 
![](https://screenshot.click/09-38-v1kez-3qar3.png)

I think we shouldn't rely on that `top` property. It seems like the svg is getting a `height: auto;` through `.card--standard:not(.card--horizontal) .placeholder-svg`. 

If we could give it back its height of `100%` then we would be good 👍 But we only need to edit that in this context. 

Maybe something like this in `template-collection.css`:
```css
.collection .media .placeholder-svg.placeholder-svg {
  height: 100%;
}
```